pyanaconda/ui/lib/subscription.py

```diff
@@ -334,10 +358,24 @@ def unregister(payload, overridden_source_type, progress_callback=None, error_ca
         # If the CDN overrode an installation source we should revert that
         # on unregistration, provided CDN is the current source.
         source_proxy = payload.get_source_proxy()
-        should_override = source_proxy.Type == SOURCE_TYPE_CDN and overridden_source_type
-        if payload.type == PAYLOAD_TYPE_DNF and should_override:
-            log.debug("subscription thread: rolling back installation source override by the CDN")
-            switch_source(payload, overridden_source_type)
+        switched_source = False
+        if payload.type == PAYLOAD_TYPE_DNF:
+            if source_proxy.Type == SOURCE_TYPE_CDN and overridden_source_type:
+                log.debug(
+                    "subscription thread: rolling back CDN installation source override"
+                )
+                switch_source(payload, overridden_source_type)
+                switched_source = True
+
+            # If requested, also restart the payload if:
+            # - installation source switch occured
+            # - the current source is CDN, which can no longer be used
+            #   after unregistration, so we need to refresh the Source
+            #   and Software spokes
+            if restart_payload and (source_proxy.Type == SOURCE_TYPE_CDN or switched_source):
+                log.debug("subscription thread: restarting payload after unregistration")
+                log.debug("AA UNREG RSTART")
```

Comment:
I guess you don't want to merge this message.

Reply:
So one debugging message did sneak through, thanks for catching it, should be fixed now. :)